### ISSUE TITLE
how can i intercept JS code before exectution in chrome

### ISSUE THREAD

**Issue by wa-sunlab** [ChromeDevTools/debugger-protocol-viewer#70]
Hi,

I would like to create chrome extension to dev tool where i would like to intercept JS code of current web page before it is compiled or executed by browser ,
Actually i want to instrument JS code before it to run in browser.

Can anybody will help us , how it is possible ?

Many Thanks in advance.

**Comment by kdzwinel** [comment]
Hey, thanks for reaching out. Unfortunately, this is not the right place to ask this question. Try SO: http://stackoverflow.com/questions/tagged/google-chrome-extension or ChromiumDev slack: https://chromiumdev-slack.herokuapp.com/?linkId=18742368

**Comment by caspervonb** [comment]
@wa-sunlab Ping me with a stackoverflow question and I'll elaborate.

[Thinamajig.js](https://github.com/hasharray/thingamajig.js) instruments with a few simple steps.
TLDR;

1. Stop the document from loading with `window.stop`
2. Fetch the document with XHR.
3. Render scripts inert with a search and replace, e.g
    
    ```js
    var html = request.responseText
      .replace(/type=\"text\/javascript\"/g, '')
      .replace(/<script/g, '<script type="live/javascript"');
    ```
4. document.write the processed html.
5. instrument and load the scripts.

**Comment by wa-sunlab** [comment]
@caspervonb 
http://stackoverflow.com/questions/40375167/how-can-i-intercept-js-code-before-exectution-in-chrome

**Comment by wa-sunlab** [comment]
@caspervonb 
Could you please explain me the step 3.

Many Thanks
